### ISSUE TITLE
NoItem 2.0.7-SNAPSHOT-193 not blocking Project Table crafting

### ISSUE THREAD

**Issue by nvx** [World-of-Tomorrow/NoItem#40]
Running Tekkit, blocking crafting of DM Blocks (126:8). Using the regular 2x2 crafting area in your inventory it successfully blocks crafting, thus confirming permissions are working.

Using PermissionsBukkit 2.0 with the node noitem.nocraft.126.8 set to true. Output of "permissions check noitem.nocraft.126.8 NVX" follows

2012-10-02 04:18:25 [INFO] Player NVX sets noitem.nocraft.126.8 to true.

Crafting a DM block in a project table however, works fine without issues, when it should be blocked.

Project Table is from RP2 PR5b2 and has id 137:3

I am using the default CustomBlocks.yml which appears to have the correct details in it for the Project Table.

debug_dump.txt: http://pastie.org/4892521

CustomBlocks.yml

``` yml
RMFurnace:
    id: 126
    data: 4
    type: furnace
    itemSlots: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
    fuelSlots: [0]
    resultSlots: [14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26]
    usesFuel: true

BlulectricFurnace:
    id: 137
    data: 4
    type: furnace
    itemSlots: [0, 1, 2, 3, 4, 5, 6, 7 , 8]
    resultSlots: [10]

ProjectTable:
    id: 137
    data: 3
    type: workbench
    resultSlots: [9]
    recipeSlots: [0, 1, 2, 3, 4, 5, 6, 7, 8]

AutomaticCraftingTableMkII:
    id: 194
    data: 1
    type: workbench
    recipeSlots: [1, 2, 3, 4, 5, 6, 7, 8, 9]
    resultSlots: [0]
    fakeRecipeItems: true
```

The issue also occurs with 2.0.6 (which is what prompted me to try 2.0.7)

Edit: Completely unrelated, but eh - You should list your downloads on BukkitDev as being compatible with 1.2.5 as well as 1.3.x so Tekkit users don't hesitate to upgrade to newer versions. I know I wasn't too sure if upgrading would just result in a bunch of NPEs causing me to hesitate before trying anything newer than 2.0.5. :) The "nohold" function works great though. :)

**Comment by just-push-it** [comment]
Wow thank you, finally a detailed bug report. I had lost all hope!

Anywho, nohold is the easy part xD blocking crafting is very difficult for custom items because they do not fire crafting events which can just be cancelled. I appreciate your taking the time to report this to me and am sorry for any trouble this is causing you.

I will look into this matter ASAP. But fair warning, if this is an issue with Tekkit itself, I can not really fix it. I only say this because there is another known issue where placing ender chests do not fire block place events or even interact events, thus can not be blocked by the plugin.

**Comment by nvx** [comment]
Ah sorry, how's this "zomg it's broken and don't work, fix plox!" - better bug report? :P

If it's an issue with Tekkit that doesn't really phase me, I'm in the process of moving to a custom launcher so we can update and add new mods as required, and I'm not adverse to manually patching classes to fix things (it wouldn't be the first time :P). So if you spot what's causing it, and it's not something you can fix, please let me know and I'll see what I can do. :)

**Comment by just-push-it** [comment]
Well I already found one issue. I have a null check on line 318 of the EventListener, which is preventing the all of the code past that from running. While doing this I also found issues with the configuration and such, so I am working on fixes for all of those and HOPEFULLY should have a patch by the end of the week. (So much to do, so little time xD )

**Comment by nvx** [comment]
haha excellent. Let me know if you need me to help test. :)

**Comment by just-push-it** [comment]
This and issue #41 should be fixed in commit d2af477fbd60f4d20ea645ad9a6308a053189cef

If I could get confirmation that would be awesome xD
You can find the build here -> http://ci.worldoftomorrow.net/job/NoItem/202/

**Comment by just-push-it** [comment]
This and #41 are FOR SURE fixed now, I tested it myself xD d419f34dad0e9a8b83c6ffc12b120ae038c926b1
